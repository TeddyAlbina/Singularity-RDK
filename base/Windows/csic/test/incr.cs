class T {
	static int[] x;
	static public void Main() {
		x = new int[5];
		x[f(1)] = x[f(2)] += f(3);
		++x[f(3)];
		x[f(4)]++;
		print(x);
	}
	static int f(int i) {
		System.Console.WriteLine(i);
		return i;
	}
	static void print(int[] x) {
		for (int i = 0; i < x.Length; i++)
			System.Console.Write(x[i]);
		System.Console.WriteLine();
	}
		
}
