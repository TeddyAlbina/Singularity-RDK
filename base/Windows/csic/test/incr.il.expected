.assembly 'incr' { }
.assembly extern 'mscorlib' {
.ver 1:0:5000:0
.publickeytoken = ( b7 7a 5c 56 19 34 e0 89 )
}
.class 'T'
	extends [mscorlib]System.Object
{
.field static private int32[] 'x'
.method hidebysig static public void 'Main'() {
.entrypoint
ldc.i4 5
newarr int32
stsfld int32[] 'T'::'x'
ldsfld int32[] 'T'::'x'
ldc.i4 1
call  int32 'T'::'f'(int32)
ldsfld int32[] 'T'::'x'
.locals init ([0]int32[] 't0')
dup
stloc 0 // t0
ldc.i4 2
call  int32 'T'::'f'(int32)
.locals init ([1]int32 't1')
dup
stloc 1 // t1
ldelem.i4
ldc.i4 3
call  int32 'T'::'f'(int32)
.locals init ([2]int32 't2')
dup
stloc 2 // t2
add
.locals init ([3]int32 't3')
stloc 3 // t3
ldloc 0 // t0
ldloc 1 // t1
ldloc 3 // t3
stelem.i4
ldloc 3 // t3
stelem.i4
ldsfld int32[] 'T'::'x'
.locals init ([4]int32[] 't4')
dup
stloc 4 // t4
ldc.i4 3
call  int32 'T'::'f'(int32)
.locals init ([5]int32 't5')
dup
stloc 5 // t5
ldelem.i4
ldc.i4.1
add
.locals init ([6]int32 't6')
stloc 6 // t6
ldloc 4 // t4
ldloc 5 // t5
ldloc 6 // t6
stelem.i4
ldsfld int32[] 'T'::'x'
.locals init ([7]int32[] 't7')
dup
stloc 7 // t7
ldc.i4 4
call  int32 'T'::'f'(int32)
.locals init ([8]int32 't8')
dup
stloc 8 // t8
ldelem.i4
ldc.i4.1
add
.locals init ([9]int32 't9')
stloc 9 // t9
ldloc 7 // t7
ldloc 8 // t8
ldloc 9 // t9
stelem.i4
ldsfld int32[] 'T'::'x'
call  void 'T'::'print'(int32[])
ret
.maxstack 5
} // end of method T.Main
.method hidebysig static private int32 'f'(int32 'i') {
.locals init ([0]int32 't0')
ldarg 0 // i
call  void [mscorlib]System.Console::'WriteLine'(int32)
ldarg 0 // i
stloc 0 // t0
br $1
$1:
ldloc 0 // t0
ret
.maxstack 1
} // end of method T.f
.method hidebysig static private void 'print'(int32[] 'x') {
.locals init ([0]int32 't0')
ldc.i4 0
stloc 0 // i
br $4
$1:
ldarg 0 // x
ldloc 0 // i
ldelem.i4
call  void [mscorlib]System.Console::'Write'(int32)
$2:
ldloc 0 // i
ldc.i4.1
add
.locals init ([1]int32 't1')
stloc 1 // t1
ldloc 1 // t1
stloc 0 // i
$4:
ldloc 0 // i
ldarg 0 // x
ldlen
conv.i4
blt $1
$3:
call  void [mscorlib]System.Console::'WriteLine'()
ret
.maxstack 2
} // end of method T.print
.method hidebysig specialname rtspecialname public void .ctor() {
ldarg.0
call instance void [mscorlib]System.Object::.ctor()
ret
.maxstack 1
} // end of constructor T.T
} // end of class T
